Mathlib/Algebra/Group/InjSurj.lean

```diff
@@ -193,9 +193,9 @@ See note [reducible non-instances]. -/
 @[reducible]
 protected def addCommMonoidWithOne {M₁} [Zero M₁] [One M₁] [Add M₁] [SMul ℕ M₁] [NatCast M₁]
     [AddCommMonoidWithOne M₂] (f : M₁ → M₂) (hf : Injective f) (zero : f 0 = 0) (one : f 1 = 1)
-    (add : ∀ x y, f (x + y) = f x + f y) (nsmul : ∀ (x) (n : ℕ), f (n • x) = n • f x)
+    (add : ∀ x y, f (x + y) = f x + f y) (nsmul : ∀ (n : ℕ) (x), f (n • x) = n • f x)
     (nat_cast : ∀ n : ℕ, f n = n) : AddCommMonoidWithOne M₁ :=
-  { hf.addMonoidWithOne f zero one add nsmul nat_cast, hf.addCommMonoid f zero add nsmul with }
+  { hf.addMonoidWithOne f zero one add nsmul nat_cast, hf.addCommSemigroup f add with }
```

Comment:
This looks like a bad change to me; it makes things more fragile to instance parent reordering.